### ISSUE TITLE
[apex] ApexCRUDViolation false negative - DML on list-indexed element not flagged

### ISSUE THREAD

**Issue by nikhil-mittal-165** [pmd/pmd#6624]
# GitHub Issue: PMD ApexCRUDViolation False Negative — DML on List-Indexed Element

**Repo:** https://github.com/pmd/pmd/issues

---

## Title

`[apex] ApexCRUDViolation false negative — DML on list-indexed element not flagged`

## Labels

`a:false-negative`, `lang:apex`

---

### Affects PMD Version

_Latest (please confirm version when filing)_

### Rule

`ApexCRUDViolation` (category: `Security`)

### Description

`ApexCRUDViolation` does not flag DML operations when the target is a **list element accessed by index** (e.g., `update myList[0]`), while correctly flagging DML on simple variable references (e.g., `insert myVariable`) in the same class and method.

### Expected Behavior

Both of the following should be flagged as CRUD violations:

```apex
insert toInsertCase;       // Correctly flagged
update accountList[0];     // Should be flagged but is NOT
```

### Actual Behavior

Only the `insert` on a simple variable is flagged. The `update` on the list-indexed element produces **no violation**.

### Minimal Reproducible Example

```apex
public with sharing class CRUDViolationReproduction {

    public static void demonstrateIssue() {
        // Setup
        List<Account> accounts = [SELECT Id, Name FROM Account LIMIT 1];
        Case myCase = new Case(Subject = 'Test');

        // DML #1: simple variable — FLAGGED correctly
        insert myCase;

        // DML #2: list-indexed element — NOT FLAGGED (false negative)
        accounts[0].Name = 'Updated Name';
        update accounts[0];
    }
}
```

**Expected:** Both `insert myCase` (line 9) and `update accounts[0]` (line 12) should trigger `ApexCRUDViolation`.

**Actual:** Only `insert myCase` is flagged.

### Customer Impact

A customer reported that their rule set (`pmd:ApexCRUDViolation`, `pmd:ApexCRUDViolation`, `pmd:ApexSharingViolations`, etc.) correctly flags an `insert` on a simple variable but does not flag an `update` on a list-indexed element (`update abnAccount[0]`) within the same method of the same class. This leads to a false sense of security compliance.
